node/src/DataConsumer.ts

```diff
@@ -565,6 +565,18 @@ export class DataConsumer<DataConsumerAppData extends AppData = AppData>
 	{
 		logger.debug('setSubchannels()');
 
+		// Optimistic subchannels update. We do this to make the |subchannels|
+		// getter be immediately updated with given values. Later it will be
+		// updated again with the response from the worker.
+		const subchannelsSet = new Set<number>();
+
+		for (const subchannel of subchannels)
+		{
+			subchannelsSet.add(Math.abs(subchannel % 65536));
+		}
+
+		this.#subchannels = Array.from(subchannelsSet).sort();
```

Comment:
Why don't we simply wait for the response to update `#subchannels`?

Reply:
Please read the PR description.

Reply:
> Also make dataConsumer.subchannels be immediately (and optimistically) updated before dataChannel.setSubchannels() completes.

I read, by I wonder *why* do we need this.

Reply:
* Some part of the app does: `dataConsumer.setSubchannels([ 1, 2, 3 ])`. It takes some times to complete.
* In the meantime `dataConsumer.subchannels` remains `[ ]` (empty).
* Another part of the app wants to add subchannel 4: `dataConsumer.setSubchannels([ ...dataConsumer.subchannels, 4 ])`.
* Result? subchannels are `[ 4 ]` instead of the intended `[ 1, 2, 3, 4 ]`.

Reply:
I see, so the only concern is that some `temporal` subchannel does not end up being a real one. I don't remember and there may be a reason for not doing it, I'm just asking: wouldn't `addSubchannels()`, `removeSubchannels()` make sense? Or is it too much clutter creating those new methods?

Reply:
> I see, so the only concern is that some temporal subchannel does not end up being a real one.

That's why it's called optimistic update, but if you look at the code it's guaranteed that it behaves fine.

> I'm just asking: wouldn't addSubchannels(), removeSubchannels() make sense? 

Too much clutter IMHO.